### ISSUE TITLE
[BUG] Triggers java.lang.IllegalStateException: Will be generated in Android

### ISSUE THREAD

**Issue by ivanovpv** [Mr-Mappie/mappie#129]
**Describe the bug**
Works well under JVM, but with Android it doesn't work

**To Reproduce**
Simplest Android unit test fails

```
import org.junit.runner.RunWith
import org.junit.Test
import tech.mappie.api.ObjectMappie

@RunWith(AndroidJUnit4::class)
class MapperTest {

    @Test
    fun testClassic() {
        val person = Person("John", "Doe", 99)
        val dto = PersonToPersonDtoMapper.map(person) //raises java.lang.IllegalStateException: Will be generated
        println("dto = $dto")
    }
}

object PersonToPersonDtoMapper : ObjectMappie<Person, PersonDto>() {
    override fun map(from: Person) = mapping {
        to::name fromValue "${from.name} ${to.name}"
    }
}

data class Person(
    val name: String,
    val surname: String,
    val age: Int,
)

data class PersonDto(
    val name: String,
    val age: Int,
)

```

**Expected behavior**
Expected translation from domain to dto bean

**Additional context**
Kotlin = 2.0.20
Android API = 34
mappieApi = "0.9.0" ("0.8.0" - the same behaviour)
For sure mappie plugin/dependencies are included.

Sources to reproduce bug: https://github.com/ivanovpv/TestApplication

**Comment by stefankoppier** [comment]
Hi @ivanovpv. Thanks for taking the time to report this issue. I will look into it.

On a side-note, I also see that the example in the README is wrong, it should be `"${from.name} ${from.name}"` instead of `"${from.name} ${to.name}"`. The example is updated. I will also create a bug report for the issue the wrong example causes

**Comment by stefankoppier** [comment]
The Gradle plugin incorrectly thinks that the mappie-api dependency is not available in the androidTest source set. A quick fix is adding `androidTestImplementation("tech.mappie:mappie-api:0.9.0")` to your build.gradle.kts file.

I will create and release a fix.

**Comment by stefankoppier** [comment]
@ivanovpv the fix for this problem is tested in your repo and is being released at this moment in version 0.9.1. Feel free to re-open this issue when you still encounter this issue.

**Comment by ivanovpv** [comment]
Great! Thanks for quick response 😊

сб, 16 нояб. 2024 г., 19:19 Stefan Koppier ***@***.***>:
